usaspending_api/awards/v2/filters/matview_filters.py

```diff
@@ -249,9 +249,9 @@ def keyword_parse(keyword):
                 queryset = queryset.filter(cfda_number__in=in_query)
 
         elif key == "naics_codes":
-            in_query = [v for v in value]
-            if len(in_query) != 0:
-                queryset = queryset.filter(naics_code__in=in_query)
+            if value:
```

Comment:
Can we add a validation to this?  It is very easy to submit requests that crash the database query which has me worried.  Turns out, you can do it in the TinyShield filter.  Search for `{"name": "naics_codes", "type": "array", "array_type": "text", "text_type": "search"}` and replace it with `{"name": "naics_codes", "type": "array", "array_type": "integer", "text_type": "search"}`.

You'll also need to update this bit here to convert all the ints to strings probably using a list comprehension `regex = f"^({'|'.join([str(v) for v in value])}).*"`.

Reply:
That looks like a good solution to this problem, but later we will need to add this functionality for TAS and PSC, which iirc will not be strictly numeric. I'm trusting Django to sanitize strings here, so can you provide an example of what might crash the DB?

Reply:
Django sanitizes the strings for SQL not for regex.  Any regex construct could potentially be used to crash the query.  Try searching for a close paren... or brace... or bracket... etc.

Reply:
Does TinyShield specify a regex for allowed inputs? That's the best way I can think of to deal with this, if our library supports it...

Reply:
I don't believe so.